scripts/build/macOS/fix_bundle.py

```diff
@@ -0,0 +1,121 @@
+import argparse
+import os
+import subprocess
+import re
+
+def is_system_lib(path):
+    return path.startswith('/System/Library/') or path.startswith('/usr/lib/')
+
+
+def macos_collect_rpaths(file):
+    result = []
+
+    with subprocess.Popen(['otool', '-l', file], stdout=subprocess.PIPE) as p:
+        lines = [line.decode('utf-8').strip() for line in p.stdout.readlines()]
+        for line_index in range(len(lines)):
+            if lines[line_index] == 'cmd LC_RPATH':
+                rpath_match = re.match(r'path\s+(.*)\s+\(', lines[line_index + 2])
+                if rpath_match:
+                    rpath = rpath_match.group(1)
+                    result.append(rpath)
+
+            line_index = line_index + 1
+
+    return result
+
+
+def get_dylib_id(file):
+    with subprocess.Popen(['otool', '-D', file], stdout=subprocess.PIPE) as p:
+        lines = [line.decode('utf-8').strip() for line in p.stdout.readlines()]
+        if len(lines) == 2:
+            return lines[1].split('/')[-1]
+    return ''
+
+
+def macos_collect_dependencies(file):
+    result = []
+    dylib_id = get_dylib_id(file)
+
+    with subprocess.Popen(['otool', '-L', file], stdout=subprocess.PIPE) as p:
+        lines = [line.decode('utf-8').strip() for line in p.stdout.readlines()]
+        for line in lines:
+            match = re.match(r'(.*)\s+\(', line)
+            if match:
```

Comment:
I think you should reduce cyclomatic complexity by returning early rather than constant nesting.